blueprint/core/blueprint_ShadowView.h

```diff
@@ -89,6 +91,54 @@ namespace blueprint
         }
     };
 
+    template <typename Setter, typename ...Args>
+    const auto getYogaNodeFloatSetter(Setter setter, Args... args) {
+      return [=](const juce::var& value, YGNodeRef node) {
+        if(value.isDouble()) {
+          setter(node, args..., (float) value);
+          return true;
+        }
+        return false;
+      };
+    }
+
+    template <typename Setter, typename SetterPercent, typename ...Args>
+    const auto getYogaNodeDimensionSetter(Setter setter, SetterPercent setterPercent, Args... args) {
+      return [=, floatSetter = getYogaNodeFloatSetter(setter, args...)](const juce::var& value, YGNodeRef node) {
+        if (floatSetter(value, node))
+          return true;
+        if (value.isString() && value.toString().trim().contains("%"))
```

Comment:
Neat! Like the macro based setter switching.

Reply:
Same! This is slick.

One quick nitpick– probably can skip the `trim()` there? It shouldn't affect finding a `"%"`